podvm-builder/podvm-builder.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+#
+
+# function to create podvm image
+
+function create_podvm_image() {
+  case "${CLOUD_PROVIDER}" in
+  azure)
+    echo "Creating Azure image"
+    /scripts/azure-podvm-image-handler.sh -c
+    if [ "${UPDATE_PEERPODS_CM}" == "yes" ]; then
+      # Get the IMAGE_ID value from the podvm-images configmap
+      # key: azure and the first value in the list
+      IMAGE_ID=$(kubectl get configmap podvm-images -n openshift-sandboxed-containers-operator -o jsonpath='{.data.azure}' | awk '{print $1}')
+      # Update peer-pods-cm configmap with the IMAGE_ID value
+      echo "Updating peer-pods-cm configmap with IMAGE_ID=${IMAGE_ID}"
+      kubectl patch configmap peer-pods-cm -n openshift-sandboxed-containers-operator --type merge -p "{\"data\":{\"AZURE_IMAGE_ID\":\"${IMAGE_ID}\"}}"
+    fi
+    ;;
+  aws)
+    echo "Creating AWS AMI"
+    /scripts/aws-podvm-image-handler.sh -c
+    if [ "${UPDATE_PEERPODS_CM}" == "yes" ]; then
+      # Get the AMI_ID value from the podvm-images configmap
+      # key: aws and the first value in the list
+      AMI_ID=$(kubectl get configmap podvm-images -n openshift-sandboxed-containers-operator -o jsonpath='{.data.aws}' | awk '{print $1}')
+      # Update peer-pods-cm configmap with the AMI_ID value
+      echo "Updating peer-pods-cm configmap with AMI_ID=${AMI_ID}"
+      kubectl patch configmap peer-pods-cm -n openshift-sandboxed-containers-operator --type merge -p "{\"data\":{\"PODVM_AMI_ID\":\"${AMI_ID}\"}}"
+    fi
+    ;;
+  *)
+    echo "CLOUD_PROVIDER is not set to azure or aws"
+    exit 1
+    ;;
+  esac
+}
+
+# Function to delete podvm image
+function delete_podvm_image() {
+  case "${CLOUD_PROVIDER}" in
+  azure)
+    echo "Deleting Azure image"
+    /scripts/azure-podvm-image-handler.sh -C
+    # Update the podvm-images configmap to remove the azure image id from the list
+    # Get the IMAGE_ID_LIST from the podvm-images configmap
+    IMAGE_ID_LIST=$(kubectl get configmap podvm-images -n openshift-sandboxed-containers-operator -o jsonpath='{.data.azure}')
+
+    # Remove the IMAGE_ID from the list
+    IMAGE_ID_LIST="${IMAGE_ID_LIST//${IMAGE_ID}/}"
+
+    # Update the podvm-images configmap with the new list
+    kubectl patch configmap podvm-images -n openshift-sandboxed-containers-operator --type merge -p "{\"data\":{\"azure\":\"${IMAGE_ID_LIST}\"}}"
+
+    ;;
+  aws)
+    echo "Deleting AWS AMI"
+    /scripts/aws-podvm-image-handler.sh -C
```

Comment:
The assumption is we never delete image gallery?

Reply:
Atleast not via the image creation/deletion job manifest. I'll include instructions to delete it. The code is there

Reply:
Example for deleting a gallery
```
apiVersion: batch/v1
kind: Job
metadata:
  name: osc-podvm-gallery-deletion
  namespace: openshift-sandboxed-containers-operator
spec:
  parallelism: 1
  completions: 1
  backoffLimit: 1
  template:
    metadata:
      name: osc-podvm-gallery-deletion
    spec:
      containers:
        - name: delete-gallery
          image: quay.io/bpradipt/osc-payload:latest
          securityContext:
            runAsUser: 0 # needed for container mode dnf access
          envFrom:
            - secretRef:
                name: peer-pods-secret
            - configMapRef:
                name: peer-pods-cm
                optional: true
            - configMapRef:
                name: azure-podvm-image-cm
                optional: true
            - configMapRef:
                name: aws-podvm-image-cm
                optional: true
          command: ["/scripts/azure-podvm-image-handler.sh", "-G", "force"]

      restartPolicy: Never
```

Reply:
WDYT about automatically delete the gallery if image list is empty?

Reply:
What if the admin wants to retain the gallery after cleanup?   Deletion is always little tricky. In docs, we can mention usage of force (if not empty) otherwise just using "-G". We can handle these in the docs and if we get feedback for automation from end users, then we can take a call. Wdyt ?

Reply:
I don't have a strong opinion about that, however, in my eyes, this all image creation idea is simply to create podvm images, the user/admin shouldn't care if v2 gallery was created behind the scenes or not, hence, it's ok to delete it if we created it.
But, again, not a big deal, you can continue as is.

Reply:
I have added another command in the podvm_builder.sh for gallery image deletion (with force option). This can be called from the same job